solr/core/src/java/org/apache/solr/response/transform/SubQueryAugmenterFactory.java

```diff
@@ -340,7 +346,9 @@ public void transform(SolrDocument doc, int docid) {
     final SolrParams docWithDeprefixed =
         SolrParams.wrapDefaults(new DocRowParams(doc, prefix, separator), baseSubParams);
     try {
-      QueryResponse rsp = server.query(coreName, docWithDeprefixed);
+      QueryRequest req = new QueryRequest(docWithDeprefixed);
+      req.setUserPrincipal(principal);
+      QueryResponse rsp = req.process(server, coreName);
```

Comment:
I'm kind of concerned this is merely one spot of potentially many other cases.  Could/should we depend on SolrRequestInfo to make this automatic?

Reply:
I commented re: SolrRequestInfo separately but will add detail there. The question is: what can we assume about the contents of SolrRequestInfo? In the context of subquery execution, what we find at the top of the stack inside SolrRequestInfo is the _new_ subquery request that we've constructed, not the enclosing/original request that contained the authentication info. The following excerpt from EmbeddedSolrServer shows how SolrRequestInfo gets set before a subquery is actually executed:

```
      req = _parser.buildRequestFrom(core, params, getContentStreams(request));
      SolrQueryResponse rsp = new SolrQueryResponse();
      SolrRequestInfo.setRequestInfo(new SolrRequestInfo(req, rsp));
      core.execute(handler, req, rsp);
```

My basic intuition is that if we're creating a brand new QueryRequest, we really _should_ be copying any authentication info onto it that needs to be used in its execution. I don't see any notion in the codebase of a parent/child relationship between queries, where a child query could somehow know who its parent is and derive auth info from the parent. Therefore, it really seems to me like copying the auth info is the logical option.